tests/README.md

```diff
@@ -6,62 +6,498 @@ Unit tests are written using [Unity](http://www.throwtheswitch.org/getting-start
 
+Finally, we can verify that everything works correctly. First, let's build the project:
 ```bash
-python3 runner.py -T ia32-generic-qemu
+TARGET=ia32-generic-qemu SYSPAGE="psh pc-ata uart16550" CONSOLE=serial ./phoenix-rtos-build/build.sh core fs project image test
+```
+Once the project is built, we are ready to run our test:
+```bash
+./phoenix-rtos-tests/runner.py --target ia32-generic-qemu -t phoenix-rtos-tests/hello/test_config.yaml
+```
+We should expect the following output:
 ```
+Flashing an image to device...
+Done!
+phoenix-rtos-tests/hello/hello-world: OK
+TESTS: 1 PASSED: 1 FAILED: 0 SKIPPED: 0
+```
+The output provides information that the device has been flashed correctly (even if it's an emulator process), followed
+by the list of tests with their. At the end, we get a summary of all the tests.
+
+Now, let's go back to the `hello_harness.py` and change the expected message:
+```python
+# We removed a comma to force a failed assertion.
+expected = "Hello world!"
+```
+Run tests one more time. Now the output should differ from the previous one:
+```
+Flashing an image to device...
+Done!
+phoenix-rtos-tests/hello/hello-world: FAIL
+ASSERTION TRACEBACK (most recent call last):
+  File "/home/user/phoenix-rtos-project/phoenix-rtos-tests/hello/hello_harness.py", line 7, in harness
+    assert line == expected, f"Line mismatch! Expected: {expected}, got: {line}" # Assure that the read line is `Hello, world!` message.
 
-The expected output is shown in the screenshot below:
+ASSERTION MESSAGE:
+Line mismatch! Expected: Hello world!, got: Hello, world!
+TESTS: 1 PASSED: 0 FAILED: 1 SKIPPED: 0
+```
+As we can see, the test finished with a failure. As a result, the runner printed more information, including a traceback
+and an assertion message. Depending on the exception type, the runner will try to provide as much information as
+possible.
 
-<img src="_images/runner-output.png" width="600px">
 
-There is also a possibility to run tests from a specified directory:
+## Example 2: unit tests using C
+In this section we will explore another type of test: unit testing. When we talk about testing, unit testing often comes
+to mind, which is why the test runner has native support for unit testing. Similar to the previous section, we start by
+creating a C file named `dummy.c` located in the `phoenix-rtos-tests/dummy` directory. To write unit tests, we will use
+the modified [Unity Test](http://www.throwtheswitch.org/unity), a thirdparty unit testing framework built for C.
 
-<img src="_images/runner-output-sample.png" width="600px">
+```c
+#include <string.h>
 
-For more runner options type:
+#include "unity_fixture.h"
 
-```bash
-python3 runner.py --help
+TEST_GROUP(dummy);
+
+TEST_SETUP(dummy) {}
+TEST_TEAR_DOWN(dummy) {}
+
+TEST(dummy, memset)
+{
+	unsigned char buffer[2] = {0, 0};
+	memset(buffer, 0xFF, 2);
+	for (int i = 0; i < 2; i++) {
+		TEST_ASSERT_EQUAL(buffer[i], 0xFF);
```

Comment:
I see that `TEST_ASSERT_EQUAL` is resolved to `TEST_ASSERT_EQUAL_INT`, it could be better to use `TEST_ASSERT_EQUAL_CHAR` or even `TEST_ASSERT_EQUAL_CHAR_ARRAY`. I know that it's only the example, but I'd rather correct this to avoid duplication.